exporter/sapmexporter/exporter.go

```diff
@@ -51,10 +53,12 @@ func newSAPMExporter(cfg *Config, params component.ExporterCreateParams) (sapmEx
 	if err != nil {
 		return sapmExporter{}, err
 	}
+
 	return sapmExporter{
-		client: client,
-		logger: params.Logger,
-		config: cfg,
+		client:  client,
+		logger:  params.Logger,
+		config:  cfg,
+		tracker: NewTracker(cfg, params),
```

Comment:
Seems better to just make this nil if correlation is disabled and then guard the `se.tracker.AddSpans` call below (or make the method handle nil receiver).  That way the tracker itself doesn't have to have all those conditions checking for nil and enabled.

Reply:
Eh, I'm not sure it's much better but I changed it. It requires added nil checks to startup/shutdown. See if you have any other ideas.